fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBQueriedRecord.java

```diff
@@ -62,12 +65,68 @@
     @Nullable
     public abstract IndexEntry getIndexEntry();
 
+    /**
+     * Get the synthetic record if this is a synthetic record type..
+     * @return a synthetic record of {@code null}
+     */
+    @Nullable
+    public abstract FDBSyntheticRecord getSyntheticRecord();
+
+    /**
+     * Get the name of this constituent if it comes from a synthetic record.
+     * @return a constituent name of {@code null}
+     */
+    @Nullable
+    public abstract String getConstituentName();
+
     /**
      * The constituents of this record if it is a synthetic record type.
      * @return a map of constituents from name to {@link FDBStoredRecord}
      */
     @Nullable
-    public abstract Map<String, FDBStoredRecord<? extends Message>> getConstituents();
+    public Map<String, FDBStoredRecord<? extends Message>> getConstituents() {
+        FDBSyntheticRecord syntheticRecord = getSyntheticRecord();
+        if (syntheticRecord == null) {
+            return null;
+        }
+        return syntheticRecord.getConstituents();
+    }
+
+    /**
+     * Get a constituent record of this record if it is a synthetic record type.
+     * @param constituentName name of the constituent
+     * @return a constituent record of {@code null}
+     */
+    @Nullable
+    @SuppressWarnings("unchecked")
+    public FDBQueriedRecord<M> getConstituent(@Nonnull String constituentName) {
```

Comment:
I guess this needs to stay to deal with the covered records that can be synthetic records? Not too happy with that ambiguity but I don't have anything better.